### PR TITLE
Adds a script to find examples of particular blocks using specific properties from Flow Cloud

### PR DESCRIPTION
**Pull request by lukestanley**
We often need to look at examples of Flows using specific blocks, and sometimes specific config properties on them - like Forms using "uiSchema". I (well, and GPT-4) made this to help make that process much faster.

### CODE REVIEW

File: docs/extract-example-block-usage-from-flows.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  let enhancedFlowsWithBlocks = [];
+  let groupedByAdapter = {};
+
+  // Iterate through the flows to collect blocks and their metadata
+  for (const flow of flowsData) {
+    let flowBlocks = [];  // To store the blocks with the property of interest for this flow
+
+    if (flow.blocks) {
+      for (const block of flow.blocks) {
+        if (block.type === blockType && (!propertyOfInterest || (block[propertyOfInterest] && Object.keys(block[propertyOfInterest]).length > 0))) {
```
Comment (gsambrotta) [inline docs/extract-example-block-usage-from-flows.js:42]: This is quite hard to read, at least for me. 
Can we at least save some of the logic in a var so is a bit more readable? 
Like `const emptyPropertOfInterest = block[propertyOfInterest] && Object.keys(block[propertyOfInterest]).length > 0`
and `const propertyOfInterestIsFalse = !propertyOfInterest || emptyPropertOfInterest`

what do you think @lukestanley ?

File: docs/extract-example-block-usage-from-flows.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      }
+    }
+
+    if (flowBlocks.length > 0) {
+      // Add metadata
+      const blockMetadata = {
+        adapterName: flow.adapterName,
+        id: flow.id,
+        title: flow.title,
+        url: `https://app.kendra.io/${flow.adapterName}/${flow.id}`
```
Comment (gsambrotta) [inline docs/extract-example-block-usage-from-flows.js:54]: Can we save static Url in a var on top of the file?
Like `const KENDRAIO_APP_URL = https://app.kendra.io/`

File: docs/extract-example-block-usage-from-flows.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+ *
+ * How to Use:
+ * 1. Modify the variables below to specify the block type and property of interest (if any).
+ * 2. Run the script using Node.js.
+ * 3. Check the generated JSON file for the grouped data.
+ */
+
+
+let inputFilePath = 'flows.json';
+let outputFilePath = 'flow-block-config-analysis.json';
```
Comment (gsambrotta) [inline docs/extract-example-block-usage-from-flows.js:17]: I don't find it clear where the files will be saved/outputted.
Also I don't think `root` is the best place. What about create a `flow-analysis` folder and keep these files there? 
And then we can hide from git so we don't deploy it.

**Review by gsambrotta — CHANGES_REQUESTED** [review]
Niiiiceee! :star2:  
This is exactly what we need! Thank you very much for working on this 
Code generally looks good to me, just some small comments about code style.

File: docs/extract-example-block-usage-from-flows.js (hunk 3 above)
Comment (lukestanley) [reply docs/extract-example-block-usage-from-flows.js]: Done @gsambrotta

**Comment by gsambrotta** [comment]
Looks good to me now :clap: :clap: 

Don't we want to add this script to the docs? We could add a section like "utility" or "helpful tools" and add this there.
What do you think @lukestanley ? I think would be a pity if this scripts stays hidden in the code (and probably get forgotten soon)

**Comment by lukestanley** [comment]
Ready for review @gsambrotta, thanks.

**Review by gsambrotta — APPROVED** [review]
Awesome!! :clap: :clap:

**Pull request merged**